keps/sig-api-machinery/2876-crd-validation-expression-language/README.md

```diff
@@ -263,35 +278,60 @@ is scoped to.
     
       It will cause a lot of keywords to be reserved and users have to memorize those variable when writing rules.
     - Using other names like `this`, `me`, `value`, `_`. The name should be self-explanatory, less chance of conflict and easy to be picked up.
-- For OpenAPIv3 object types, the expression will have direct access to all the
-fields of the object the validator is scoped to.
+
+- For OpenAPIv3 object types, the expression may use field selection to access all the
+  properties of the object the validator is scoped to, e.g. `self.field == 10`.
   
 - For OpenAPIv3 scalar types (integer, string & boolean), the expression will have access to the
-scalar data element the validator is scoped to. The data element will be accessible to CEL
-expressions via `self`, e.g. `len(self) > 10`.
+  scalar data element the validator is scoped to. The data element will be accessible to CEL
+  expressions via `self`, e.g. `len(self) > 10`.
 
 - For OpenAPIv3 list and map types, the expression will have access to the data element of the list
 or map. These will be accessible to CEL via `self`. The elements of a map or list can be validated using the CEL support for collections
 like the `all` macro, e.g. `self.all(listItem, <predicate>)` or `self.all(mapKey,
 <predicate>)`.
   
 - For immutability use case, validator will have access to the existing version of the object. This
-  will be accessible to CEL via the `old<propertyName>` identifier.
+  will be accessible to CEL via the `oldSelf` identifier.
   - This will only be available on mergable collection types such as objects (unless
     `x-kubernetes-map-type=atomic`), maps with `x-kubernetes-map-type=granular` and lists
     with `x-kubernetes-list-type` set to `set` or `map`.  See [Merge
     Strategy](https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy) for
     details.
   - The use of "old" is congruent with how `AdmissionReview` identifies the existing object as
-    `oldObject`. To avoid name collisions `old<propertyName>` will be treated the same as a CEL
-    keyword for escaping purposes (see below).
+    `oldObject`.
   - xref [analysis of possible interactions with immutability and
     validation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1101-immutable-fields#openapi-extension-x-kubernetes-immutable).
 
-- If a object property collides with a CEL keyword (see RESERVED in [CEL Syntax](https://github.com/google/cel-spec/blob/master/doc/langdef.md#syntax)),
-  it will be escaped by prepending a _ prefix. To prevent this from causing a subsequent collision, all properties with a `_` prefix will always be
+- If a object property name is a CEL keyword (see RESERVED in [CEL Syntax](https://github.com/google/cel-spec/blob/master/doc/langdef.md#syntax)),
```

Comment:
Why are these escaped differently than the "reserved language identifier" down below?

Reply:
I've removed the below about reserved language identifiers. It's no longer relevant now that we access all fields via `self`. I'm pretty happy about this. It really future proofs the API surface.

Reply:
I'm confused here as well.

This line says "If a object property name is a CEL keyword (see RESERVED... ) it will be escaped by prepending a _ prefix."

A few lines down from here says "Only property names of the form `[a-zA-Z_.-/][a-zA-Z0-9_.-/]*` are accessible and are escaped according to the following rules... Property names that match a CEL RESERVED keyword exactly escape to '\__{keyword}__'."

These seem to refer to the same RESERVED keywords and/but lay out different spellings for them?